cpp/src/phonenumbers/phonenumbermatcher.cc

```diff
@@ -628,6 +627,7 @@ bool PhoneNumberMatcher::ExtractMatch(const string& candidate, int offset,
 bool PhoneNumberMatcher::HasNext() {
   if (state_ == NOT_READY) {
     PhoneNumberMatch temp_match;
+    try {
```

Comment:
For style reasons, we don't use exceptions in C++.

Reply:
Thanks Lara..
Sorry, earlier went ahead with the suggestion as we mistakenly assumed that we are not able to replicate internally and something to do with the flags like "-fsanitize=address,fuzzer-no-link" as we see lot memory leak warnings when run that way. We were able to reproduce the issue internally as well. cl/416205516

I see we hit this issue only when the input is large (> 2000 chars) and the index going negative  in next iteration. I updated the code to check the range of index. Will debug more and see the root cause and have proper fix (like having a cap on the input length / using unsigned int for index...

Reply:
Roughly till length 3500 we are not hitting the issue. As I am not familiar with CPP, could not debug why the index becomes negative. Maybe we can go ahead with the present solution?

Reply:
Done. Updated relevant .h file. Please verify whether it fine..